sw/device/tests/sysrst_ctrl_inputs_test.c

```diff
@@ -0,0 +1,154 @@
+// kTestPhase is 0, the very first OTTF_WAIT_FOR could succeed before the host
+// can set the pins. To avoid this, and only on real devices, set the initial
+// value to an invalid value so that we have to wait for the host.
+static volatile uint8_t kTestPhaseReal = 0xff;
+static volatile uint8_t kTestExpectedReal = 0;
+
+enum {
+  kOutputNumPads = 0x8,
+  kOutputNunMioPads = 0x6,
+};
+
+static const dif_pinmux_index_t kPeripheralInputs[] = {
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonKey0In,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonKey1In,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonKey2In,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonPwrbIn,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonAcPresent,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonLidOpen,
+};
+
+static const dif_pinmux_index_t kInputPadsDV[] = {
+    kTopEarlgreyPinmuxInselIob3, kTopEarlgreyPinmuxInselIob6,
+    kTopEarlgreyPinmuxInselIob8, kTopEarlgreyPinmuxInselIor13,
+    kTopEarlgreyPinmuxInselIoc7, kTopEarlgreyPinmuxInselIoc9,
+};
+
+static const dif_pinmux_index_t kInputPadsReal[] = {
+    kTopEarlgreyPinmuxInselIor2, kTopEarlgreyPinmuxInselIor3,
+    kTopEarlgreyPinmuxInselIor4, kTopEarlgreyPinmuxInselIor5,
+    kTopEarlgreyPinmuxInselIor6, kTopEarlgreyPinmuxInselIor7,
+};
+
+static const dif_sysrst_ctrl_pin_t kSysrstCtrlInputs[] = {
+    kDifSysrstCtrlPinKey0In,           kDifSysrstCtrlPinKey1In,
+    kDifSysrstCtrlPinKey2In,           kDifSysrstCtrlPinPowerButtonIn,
+    kDifSysrstCtrlPinAcPowerPresentIn, kDifSysrstCtrlPinLidOpenIn,
+    kDifSysrstCtrlPinEcResetInOut,     kDifSysrstCtrlPinFlashWriteProtectInOut,
+};
+
+static uint8_t read_input_pins(void) {
+  bool input_value;
+  uint8_t inputs = 0;
+  for (int i = 0; i < kOutputNumPads; ++i) {
+    CHECK_DIF_OK(dif_sysrst_ctrl_input_pin_read(
+        &sysrst_ctrl, kSysrstCtrlInputs[i], &input_value));
+    inputs |= input_value << i;
+  }
+  return inputs;
+}
+
+void setup_dio_pins(dif_pinmux_t *pinmux, dif_sysrst_ctrl_t *sysrst_ctrl) {
+  // Make sure the the DIO pins EC reset and flash WP are configured correctly.
+  // The documentation says that they should be configured as open drain but on
+  // FPGA targets, only virtual open drain is supported.
+  dif_pinmux_pad_attr_t out_attr;
+  dif_pinmux_pad_attr_t in_attr = {
+      .slew_rate = 0,
+      .drive_strength = 0,
+      // FIXME FPGA requires virtual open drain, should be open drain on real
+      // chip?
+      .flags = kDifPinmuxPadAttrVirtualOpenDrain,
+  };
+  CHECK_DIF_OK(dif_pinmux_pad_write_attrs(
+      pinmux, kTopEarlgreyDirectPadsSysrstCtrlAonEcRstL, kDifPinmuxPadKindDio,
+      in_attr, &out_attr));
+  CHECK_DIF_OK(dif_pinmux_pad_write_attrs(
+      pinmux, kTopEarlgreyDirectPadsSysrstCtrlAonFlashWpL, kDifPinmuxPadKindDio,
+      in_attr, &out_attr));
+  // We also need to disable the output override mecanism (i.e. "release the
+  // pin").
+  dif_sysrst_ctrl_pin_config_t cfg = {
+      .enabled = false,
```

Comment:
Oops!

```suggestion
      .enabled = kDifToggleDisabled,
```

The API:
https://github.com/lowRISC/opentitan/blob/755180702e6e410ad895bf48c06bbce571670502/sw/device/lib/dif/dif_sysrst_ctrl.h#L452